prover/circuits/prove.go

```diff
@@ -86,7 +86,6 @@ func ProveCheck(setup *Setup, assignment frontend.Circuit, opts ...any) (plonk.P
 			assignment,
 			assignment,
 			setup.Circuit.Field(),
-			test.WithBackendProverOptions(proverOpts...),
```

Comment:
@ivokub argues that removing this is okay because it is only ever used for hint substitution, which would not be necessary for the test engine.